core/shared/platform/linux-sgx/sgx_ipfs.c

```diff
@@ -291,8 +291,31 @@ ipfs_fopen(int fd, const char *filename, int flags)
         return NULL;
     }
 
-    // Opening the file
-    void *sgx_file = sgx_fopen_auto_key(filename, mode);
+    // Determine the symbolic link of the file descriptor, because IPFS does not
+    // support opening a relative path to a file descriptor (i.e., openat).
+    // Using the symbolic link in /proc/self allows to retrieve the same path as
+    // opened by the initial openat and respects the chroot of WAMR.
+    int size_of_path = snprintf(NULL, 0, "/proc/self/fd/%d", fd) + 1;
+    if (size_of_path < 0) {
+        errno = EINVAL;
+        return NULL;
+    }
+
+    char *symbolic_path = BH_MALLOC(size_of_path);
+    snprintf(symbolic_path, size_of_path, "/proc/self/fd/%d", fd);
+
+    // Resolve the symbolic link to real absolute path, because IPFS can only
+    // open a file with a same file name it was initially created. Otherwise,
+    // IPFS throws SGX_ERROR_FILE_NAME_MISMATCH.
+    char real_path[PATH_MAX];
+    size_t ret = readlink(symbolic_path, real_path, PATH_MAX);
+    real_path[PATH_MAX - 1] = '\0';
```

Comment:
Seems not a good choice, the actual length should be much smaller than `PATH_MAX - 1`, how about `char real_path[PATH_MAX] = { 0 };` and removing this line?

Reply:
Indeed, I removed that line and initialized the array with zeroes. Furthermore, since we no longer set the last byte of the array after the syscall, I subtract the `bufsiz` of 1 when calling `readlink`, since the syscall does not append a terminating null byte.